src/main/java/CancelApplication/CancelApplicationDsGateway.java

```diff
@@ -0,0 +1,32 @@
+package CancelApplication;
+
+import Entities.Group;
+import Entities.User;
+
+public interface CancelApplicationDsGateway {
+    boolean groupExists(String groupname);
+    User getUser(String username);
+    Group getGroup(String groupname);
+
+    /**
+     * Updates the current user after user's applications list has changed. Saves to repository.
+     * @param username the username of the current user
+     */
+    void updateUser(String username);
```

Comment:
I mentioned this for your other use case, but how will we know what to update in the User? I am a little confused

Reply:
same as other usecase, thank you for catching!